opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -166,16 +166,40 @@
         SpanBuilder withStartTimestamp(long microseconds);
 
         /**
+         * @deprecated use {@link #startActive} instead.
          * Returns a newly started and activated {@link Scope}.
          *
          * <p>
-         * The returned {@link Scope} supports try-with-resources. For example:
+         * Note: {@link SpanBuilder#startActive(boolean)} is a shorthand for
+         * {@code tracer.scopeManager().activate(spanBuilder.start(), finishSpanOnClose)}.
+         *
+         * @param finishSpanOnClose whether span should automatically be finished when {@link Scope#close()} is called
+         * @return a {@link Scope}, already registered via the {@link ScopeManager}
+         *
+         * @see ScopeManager
+         * @see Scope
+         */
+        @Deprecated
+        Scope startActive(boolean finishSpanOnClose);
+
+        /**
+         * Returns a newly started and activated {@link Scope}.
+         *
+         * The span will not be automatically finished when {@link Scope#close()} is called.
+         *
+         * <p>
+         * The returned {@link Scope} supports try-with-resources, but is usually
```

Comment:
Why is it not usually used with try-with-resources?

Reply:
That's related to Yuri's comment about removing it - essentially, that, as opposed to using `ScopeManager.activate()`, here we **need** an usual `try` block to do the cleanup (that is, having a reference to the `Span` to finish it after any error has occurred).

```java
try (Scope scope = tracer.buildSpan("foo").startActive()) {
} catch (Exception e) {
  // scope and scope.span() are not reachable here nor in the finally section :/
}
```

Reply:
If this method isn't deleted, I think it would be cleaner if done similar to census and brave's "scoped span" Instead of messing with scope, mess with span.finish(). In other words, return a Span, and do mention it is a programming error to not call span.finish on the same thread. Another way is to leave it out..

Here's the javadoc from brave's ScopedSpan:

```java
  /**
   * Closes the {@link CurrentTraceContext#newScope(TraceContext) scope} associated with this span,
   * then reports the span complete, assigning the most precise duration possible.
   */
  public abstract void finish();
```